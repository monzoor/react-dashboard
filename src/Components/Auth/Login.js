import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Link, withRouter } from 'react-router-dom';
import {
    Form, Icon, Input,
    Button,
    Checkbox,
    message,
} from 'antd';

import ErrorBoundary from '../../ErrorBoundary/ErrorBoundary';
import ErrorThrower from '../../ErrorBoundary/ErrorThrower';
import { auth } from '../../Actions/authAction';

const FormItem = Form.Item;

class Login extends Component {
    static propTypes = {
        form: PropTypes.oneOfType([PropTypes.object]).isRequired,
    };

    state = {
        loading: false,
    }


    componentWillReceiveProps(nextProps) {
        const { errors } = this.props;
        if (nextProps.errors !== errors) {
            if (nextProps.errors.errorInfos.hasErrors) {
                this.setState({
                    loading: false,
                });
                message.destroy();
            }
            if (nextProps.errors.errorInfos.messages && nextProps.errors.errorInfos.componentError) {
                message.error(nextProps.errors.errorInfos.messages);
            }
        }
    }

    handleSubmit = (e) => {
        e.preventDefault();
        const { form, dispatch } = this.props;
        form.validateFields((err, values) => {
            if (!err) {
                this.setState({
                    loading: true,
                });
                dispatch(auth(this.props, values));
            }
        });
    }

    render() {
        const { loading } = this.state;
        const { form, errors } = this.props;
        const { getFieldDecorator } = form;

        return (
            <Form onSubmit={this.handleSubmit} id="logIn">
                <h1 className="text-center font-weight-light mb-4">Login</h1>
                <FormItem className="form-group">
                    {
                        getFieldDecorator('email', {
                            rules: [{
                                required: true,
                                type: 'email',
                            }],
                    })(
                        <Input prefix={<Icon type="user" className="text-muted" />} placeholder="Email" />,
                    )}
                </FormItem>
                <FormItem className="form-group">
                    {
                        getFieldDecorator('password', {
                            rules: [{
                                required: true,
                                min: 3,
                            }],
                    })(
                        <Input prefix={<Icon type="lock" className="text-muted" />} type="password" placeholder="Password" />,
                    )}
                </FormItem>
                <FormItem>
                    {
                        getFieldDecorator('remember', {
                            valuePropName: 'checked',
                            initialValue: true,
                    })(
                        <Checkbox>Remember me</Checkbox>,
                    )}
                    <Link className="float-right" to="/">Forgot password</Link>
                    <Button className="btn btn-primary" htmlType="submit" loading={loading} block>
                        Log in
                    </Button>
                    <ErrorBoundary>
                        <ErrorThrower {...errors} errorMessage="Crushed in login" />
                    </ErrorBoundary>
                    <Link to="/signup" className="btn btn-block btn-outline-primary mt-3">
                        Signup
                    </Link>
                </FormItem>
            </Form>
        );
    }
}

// const mapStateToProps = (state) => {
//     const { errors } = state;
//     return {
//         errors,
//     };
// };
const mapStateToProps = state => ({
    errors: state.errors,
});

const LoginForm = Form.create()(Login);

export default withRouter(connect(mapStateToProps)(LoginForm));
// export default WrappedNormalLoginForm;
