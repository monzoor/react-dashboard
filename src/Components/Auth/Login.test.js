// import Login from '../Home/index';
// import { Login } from './Login';
// import { BrowserRouter as Router } from 'react-router-dom';
// import ReactDOM from 'react-dom';
// import { MemoryRouter } from 'react-router';
import { Login } from './Login';
import { Form } from 'antd';
// import NotFoundPage from '../404';
// import App from '../App';

describe('This is login test', () => {
    // it('add correctly', () => {
    //     expect(1 + 1).toEqual(2);
    // });
    const fakeEvent = { preventDefault: () => console.log('preventDefault') };
    // const loginComponent = shallow(<LoginForm />);
    it('found login from', () => {
        // const wrapper = mount(
        //     <MemoryRouter initialEntries={['/random']}>
        //         <App />
        //     </MemoryRouter>,
        // );
        // expect(wrapper.find(LoginForm)).toHaveLength(0);
        // expect(wrapper.find(NotFoundPage)).toHaveLength(1);
        // const wrapper = mount(<Router><LoginForm /></Router>);
        // const push = jest.fn();
        // wrapper.setProps({ history: { push } });
        // const LoginTest = Form.create()(Login);
        // Form.setFieldsValue({
        //     email: {
        //         value: 'asdasd@f.com',
        //     },
        // });
        // const loginComponent = shallow(<LoginForm wrappedComponentRef={(form) => this.form = form} />);
        let formRef= {
            form: {},
        };
        const loginComponent = shallow(<Login{...formRef} />);
        // console.log(loginComponent.instance().props.wrappedComponentRef());
        // loginComponent.getDOMNode().setFieldsValue({
        //     name: 'tom'
        //   });
        // expect(loginComponent.node.getFieldsValue()).toEqual({name: 'tom'});
        // console.log()
        expect(loginComponent.find('#logIn').length).toBe(1);
        // loginComponent.find('#logIn').simulate('submit', fakeEvent);
        // const loginComponent = shallow(<LoginForm />).dive();
        // console.log(loginComponent.debug());
        // // expect(loginComponent.find('#logIn')).to.have.lengthOf(1);
        // // expect(loginComponent.find('#logIn').length).toBe(1);
        // expect(loginComponent.findWhere(node => node.hasClass('test'))).toBe(1);
    });
    it('User form submit test', () => {
        let formRef= {
            form: {
                setFields: {
                    email: {
                        value: 'asdasd',
                    },
                },
                validateFields: () => {},
            },
        };
        const loginComponent = shallow(<Login {...formRef} />);
        loginComponent.find('#logIn').simulate('submit', fakeEvent);
    });
    it('Noti test', () => {
        let formRef= {
            form: {
                setFields: {
                    email: {
                        value: 'asdasd',
                    },
                },
                validateFields: () => {},
                getFieldProps: (props) => {},
            },
        };
        const loginComponent = shallow(<Login {...formRef} />);
        // done// expect(loginComponent.find(Notification).length).toBe(1);
        // done// expect(loginComponent.exists()).toBe(true);
        loginComponent.find('Email').simulate('change', {
            target: { value: 'oka'}
        })
        console.log(loginComponent.find('Email'));
        expect(loginComponent.find('Email').props().form.getFieldProps('email').value).toBe('oka');
    });        
});
