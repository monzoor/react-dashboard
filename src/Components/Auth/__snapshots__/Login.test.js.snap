// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`This is login test should render correctly in "debug" mode 1`] = `
<Form
  hideRequiredMark={false}
  layout="horizontal"
  onSubmit={[Function]}
  prefixCls="ant-form"
>
  <h1
    className="text-center font-weight-light mb-4"
  >
    Login
  </h1>
  <Email
    form={
      Object {
        "getFieldDecorator": [Function],
        "getFieldError": [Function],
        "getFieldInstance": [Function],
        "getFieldProps": [Function],
        "getFieldValue": [Function],
        "getFieldsError": [Function],
        "getFieldsValue": [Function],
        "isFieldTouched": [Function],
        "isFieldValidating": [Function],
        "isFieldsTouched": [Function],
        "isFieldsValidating": [Function],
        "isSubmitting": [Function],
        "resetFields": [Function],
        "setFields": [Function],
        "setFieldsInitialValue": [Function],
        "setFieldsValue": [Function],
        "submit": [Function],
        "validateFields": [Function],
        "validateFieldsAndScroll": [Function],
      }
    }
  />
  <Password
    form={
      Object {
        "getFieldDecorator": [Function],
        "getFieldError": [Function],
        "getFieldInstance": [Function],
        "getFieldProps": [Function],
        "getFieldValue": [Function],
        "getFieldsError": [Function],
        "getFieldsValue": [Function],
        "isFieldTouched": [Function],
        "isFieldValidating": [Function],
        "isFieldsTouched": [Function],
        "isFieldsValidating": [Function],
        "isSubmitting": [Function],
        "resetFields": [Function],
        "setFields": [Function],
        "setFieldsInitialValue": [Function],
        "setFieldsValue": [Function],
        "submit": [Function],
        "validateFields": [Function],
        "validateFieldsAndScroll": [Function],
      }
    }
  />
  <div
    className="row"
  >
    <div
      className="col-6"
    >
      <RememberMe
        className="float-left"
        form={
          Object {
            "getFieldDecorator": [Function],
            "getFieldError": [Function],
            "getFieldInstance": [Function],
            "getFieldProps": [Function],
            "getFieldValue": [Function],
            "getFieldsError": [Function],
            "getFieldsValue": [Function],
            "isFieldTouched": [Function],
            "isFieldValidating": [Function],
            "isFieldsTouched": [Function],
            "isFieldsValidating": [Function],
            "isSubmitting": [Function],
            "resetFields": [Function],
            "setFields": [Function],
            "setFieldsInitialValue": [Function],
            "setFieldsValue": [Function],
            "submit": [Function],
            "validateFields": [Function],
            "validateFieldsAndScroll": [Function],
          }
        }
      />
    </div>
    <div
      className="col-6"
    >
      <Link
        className="float-right mb-2"
        replace={false}
        to="/"
      >
        Forgot password
      </Link>
    </div>
  </div>
  <Button
    block={true}
    className="btn btn-primary"
    ghost={false}
    htmlType="submit"
    loading={false}
    prefixCls="ant-btn"
  >
    Log in
  </Button>
  <withRouter(Connect(ErrorBoundary))>
    <ErrorThrower
      errorInfos={
        Object {
          "hasErrors": false,
        }
      }
      errorMessage="Crushed in login"
    />
  </withRouter(Connect(ErrorBoundary))>
  <Link
    className="btn btn-block btn-outline-primary mt-3"
    replace={false}
    to="/signup"
  >
    Signup
  </Link>
</Form>
`;
